CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+- [Picking an issue](#picking-an-issue)
+- [Installation](#installation)
+- [Writing and Running Tests](#writing--running-tests)
+- [Checking for Linter](#checking-for-linter)
+- [Checking for Prettier](#checking-for-prettier)
+- [Submitting a Bug or Issue or Feature Request](#submitting-a-bug-or-issue-or-feature-request)
+- [Creating a Pull Request](#creating-a-pull-request)
+  - [Writing Commit Message](#writing-commit-message)
+
+## Project Architecture
+
+Extension Activity Monitor is composed by a backend part (running in the background page) and a frontend part (composed by a browserAction Popup and an Activity Log extension page which is loaded as a tab or as a devtools panel).
+
+### Some Important Components
+
+#### Background Page
+
+The background page acts as the backend of the extension. The core part of the background page is implemented in [`ext-monitor.js`](https://github.com/mozilla/extension-activity-monitor/blob/master/src/lib/ext-monitor.js). It is responsible for monitoring extensions, storing activity logs, saving logs to a JSON file and loading logs from JSON file. The logs remain alive in the background unless it receives the [`clearlogs`](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L106) instruction from Activity Log page(frontend part). The background page also [sends the real-time logs](https://github.com/mozilla/extension-activity-monitor/blob/68d51940f1db397a0972658622bbdd39041436a7/src/lib/ext-monitor.js#L25) to Activity Log page. The background page communicates with popup and activity log page (which are part of frontend) more often to make sure monitoring runs smoothly.
+
+#### Popup Page
+
+The popup page is one part of the frontend that communicates with the background page (backend part) to run most of its functionality. It is responsible for "start monitoring" and "stop monitoring" other installed extensions. It also helps to access the Activity Log page.
+
+#### Activity Log Page - Tab (Extension Page)
+
+The Activity Log page is another part of the frontend. It uses the MVC architecture and [web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component) for table view and filtering options.
+
+- The **role of Model** is to store the activity logs from background and store the filters. It also checkes if logs are matching or not matching with the applied filters.
+
+- The **role of View** is to deal with DOM. It interacts with DOM whenver our Model is updated. The `View` deals with `log-view` web component to render the logs and [filter web components](https://github.com/mozilla/extension-activity-monitor/tree/master/src/lib/web-component) to display filters.
+
+- The **role of Controller** is to make sure that both Model and View are synchronized. Whenever the Model updates with new logs or filter options, the Controller make sure that the View also updates accordingly.
```

Comment:
The Controller does connect the Model to the View, and it does also:
- parse the URL search params
- interact with the backend (through WebExtensions messaging API),

Reply:
Should we differ including the `parse the URL search params` after it is being added ?